Add properties to user resource for cert creation

daviddjh opened the pull request:
# Description

This change adds properties to the user resource of the openvpn cookbook. As a result, developers can specifically define attributes of the user certificate they are trying to create within a resource statement. Environment variables are updated in the server recipe to provide default values.

daviddjh committed:
Add properties to user resource for cert creation

Signed-off-by: David Headrick <daviddjh@osuosl.org>

ramereth commented:
@daviddjh I think this looks good? Go ahead and make an upstream PR but don't forget to update the changelog

daviddjh committed:
Add configurable organizational unit within openssl template

Signed-off-by: David Headrick <daviddjh@osuosl.org>

daviddjh committed:
Update Changelog

Signed-off-by: David Headrick <daviddjh@osuosl.org>

daviddjh committed:
Added tests for user resource

Signed-off-by: David Headrick <daviddjh@osuosl.org>

daviddjh committed:
Style and test fixes

Signed-off-by: David Headrick <daviddjh@osuosl.org>

daviddjh committed:
Change server_verification test to match InSpec convention

Signed-off-by: David Headrick <daviddjh@osuosl.org>

kitchen-porter committed:
Automated PR: Cookstyle Changes (#199)

* Cookstyle 7.30.1 Fixes

Issues found and resolved with: recipes/server.rb

 - 88:3 refactor: Chef/RedundantCode/UseCreateIfMissing - Use the :create_if_missing action instead of not_if with a ::File.existFOO check. https://docs.chef.io/workstation/cookstyle/chef_redundantcode_usecreateifmissing
 - 98:3 refactor: Chef/RedundantCode/UseCreateIfMissing - Use the :create_if_missing action instead of not_if with a ::File.existFOO check. https://docs.chef.io/workstation/cookstyle/chef_redundantcode_usecreateifmissing

Signed-off-by: kitchen-porter <board@sous-chefs.org>

* Update CHANGELOG.md

Co-authored-by: Dan Webb <dan.webb@damacus.io>

kitchen-porter committed:
Update metadata for 5.4.0

kitchen-porter committed:
Update changelog for 5.4.0

kitchen-porter committed:
Set Changelog to ## Unreleased

daviddjh committed:
Condence key variables to key_vars hash...
... convert bash resources to execute resources

Signed-off-by: David Headrick <daviddjh@osuosl.org>

daviddjh committed:
Replace ternaries with ORs in user resource

kitchen-porter committed:
Automated PR: Standardising Files (#200)

* Standardise files with files in sous-chefs/repo-management

Signed-off-by: kitchen-porter <board@sous-chefs.org>

* Remove delivery and move to calling RSpec directly via a reusable workflow

Signed-off-by: Lance Albertson <lance@osuosl.org>

* Update tested platforms

Signed-off-by: Lance Albertson <lance@osuosl.org>

* Enable unified_mode and require Chef >= 15.3

Standardize kitchen settings

Signed-off-by: Lance Albertson <lance@osuosl.org>

* Fix various idempotency issues

Signed-off-by: Lance Albertson <lance@osuosl.org>

Co-authored-by: Lance Albertson <lance@osuosl.org>

kitchen-porter committed:
Update metadata for 6.0.0

kitchen-porter committed:
Update changelog for 6.0.0

kitchen-porter committed:
Set Changelog to ## Unreleased
